gax-java/gax/src/test/java/com/google/api/gax/retrying/AbstractRetryingExecutorTest.java

```diff
@@ -222,29 +223,32 @@ public void testTotalTimeoutExceeded() throws Exception {
   }
 
   @Test
-  public void testCancelOuterFutureBeforeStart() throws Exception {
-    FailingCallable callable = new FailingCallable(4, "request", "SUCCESS", tracer);
+  public void testCancelOuterFutureBeforeStart() {
+    // Use MockCallable instead of FailingCallable as we do not need the request to be re-tried.
+    // For this test, the callable should attempt to run and then see that the external future
+    // is cancelled and that the callable's execution has ended.
+    MockCallable callable = new MockCallable(tracer, "request", "SUCCESS");
 
     RetrySettings retrySettings =
         FAST_RETRY_SETTINGS
             .toBuilder()
             .setInitialRetryDelay(Duration.ofMillis(1_000L))
             .setMaxRetryDelay(Duration.ofMillis(1_000L))
-            .setTotalTimeout(Duration.ofMillis(10_0000L))
+            .setTotalTimeout(Duration.ofMillis(10_000L))
             .build();
     RetryingExecutorWithContext<String> executor =
         getExecutor(getAlgorithm(retrySettings, 0, null));
     RetryingFuture<String> future = executor.createFuture(callable, retryingContext);
     boolean res = future.cancel(false);
-
     assertTrue(res);
+    assertTrue(future.isCancelled());
 
+    callable.setExternalFuture(future);
     future.setAttemptFuture(executor.submit(future));
-
-    assertFutureCancel(future);
     assertEquals(0, future.getAttemptSettings().getAttemptCount());
 
-    verifyNoMoreInteractions(tracer);
+    // Assert that no tracer attempt has run
+    verifyNoInteractions(tracer);
```

Comment:
I wonder how useful is this verification now as we are basically verifying the behavior of our own `MockCallable`. As you mentioned in other comments, we are always checking that [the future is not done](https://github.com/googleapis/gapic-generator-java/blob/3f6c4c04fde254cfc3c1441b44beeb2bbe0c7e03/gax-java/gax/src/main/java/com/google/api/gax/retrying/CallbackChainRetryingFuture.java#L136-L138) before submitting it to an retrying executor, so the current flakiness is actually expected since we are not checking if the future is done again in the [submit](https://github.com/googleapis/gapic-generator-java/blob/3f6c4c04fde254cfc3c1441b44beeb2bbe0c7e03/gax-java/gax/src/main/java/com/google/api/gax/retrying/ScheduledRetryingExecutor.java#L115-L119) method itself. 
That being said, I don't think we want to mimic the behavior in `MockCallable` as that's outside the scope of the unit tests of  `AbstractRetryingExecutorTest`. My suggestion is that we keep this test case but remove this line of verification if we don't find any other benefit.

Reply:
MockCallable's tracer is to mimic the functionality inside AttemptCallable (which is what I believe FailingCallable is based on): https://github.com/googleapis/gapic-generator-java/blob/4bb0a5e3ce78878841e87a47cd457995ea9ed87e/gax-java/gax/src/main/java/com/google/api/gax/rpc/AttemptCallable.java#L78-L85

The tracer's functionality for the MockCallable can also be covered with the assertion below: `assertEquals(0, future.getAttemptSettings().getAttemptCount());`

so I'm fine with removing.